lib/object-mask.js

```diff
@@ -396,34 +405,30 @@ function invert(mask) {
 	return result;
 }
 
-// Throws on (deep) attempt to subtract collection from scalar
 function subtract(a, b) {
 	// base cases:
+	if (a === true) return invert(b);
+	if (a === false) return false;
+	if (!a) return invert(b);
 	if (!b) return a;
-	if (!a || b === true || _.isEqual(a, b)) return false;
-	if (objtools.isScalar(a) && !objtools.isScalar(b)) {
```

Comment:
We still need a scalar check somewhere.  If an invalid value (say, `3`) is passed in, it could otherwise get through to the code below where it tries to iterate over `a` and `b`, and crash because the value isn't iterable.